src/psyclone/psyir/symbols/symboltable.py

```diff
@@ -221,19 +222,38 @@ def specify_argument_list(self, argument_symbols):
         self._validate_arg_list(argument_symbols)
         self._argument_list = argument_symbols[:]
 
-    def lookup(self, name):
+    def lookup(self, name, visibility=None):
         '''
         Look up a symbol in the symbol table.
 
         :param str name: name of the symbol.
+        :param visibilty: the visibility or list of visibilities that the \
+                          symbol must have.
+        :type visibility: [list of] :py:class:`psyclone.symbols.Visibility`
 
-        :returns: symbol with the given name.
+        :returns: symbol with the given name and, if specified, visibility.
         :rtype: :py:class:`psyclone.psyir.symbols.Symbol`
 
+        :raises SymbolError: if the name exists in the Symbol Table but does \
+                             not have the specified visibility.
         :raises KeyError: if the given name is not in the Symbol Table.
+
         '''
         try:
-            return self._symbols[self._normalize(name)]
+            symbol = self._symbols[self._normalize(name)]
+            if visibility:
+                if not isinstance(visibility, list):
+                    vis_list = [visibility]
+                else:
+                    vis_list = visibility
```

Comment:
Codecov says this line is uncovered in the tests

Reply:
Oops. As requested, I've added a new test to check the visibility-filtering functionality.